java/java.hints/src/org/netbeans/modules/java/hints/errors/CreateElement.java

```diff
@@ -101,6 +101,17 @@ public final class CreateElement implements ErrorRule<Void> {
     private static final int PRIO_TESTSOURCEGROUP = 500;
     private static final int PRIO_MAINSOURCEGROUP = 1000;
     private static final int PRIO_INNER = 2000;
+
+    /**
+     * Helps to prevent issue 8594, trying to create a method when return type is not given 
+     * by using var as the placeholder for the type at the call site.
+     * @param type the TypeMirror that specifies the type info
+     * @return true if fit for create method, false otherwise
+     */
+    private static boolean hasKnownReturnType(TypeMirror type) {
+        if (type==null) return true; // for constructors
+        return type.getKind() != TypeKind.NONE; // none is insufficient info
+    }
```

Comment:
I suggest slightly altered wording and fixing the missing braces for the if construct

```suggestion
    /**
     * Helps to prevent github issue 8594, trying to create a method when return
     * type can't be determined because var is used as placeholder for the type
     * at the call
     * site.
     * @param type the TypeMirror that specifies the type info
     * @return true if fit for create method, false otherwise
     */
    private static boolean hasKnownReturnType(TypeMirror type) {
        // for constructors
        if (type == null) {
            return true;
        }
        return type.getKind() != TypeKind.NONE; // none is insufficient info
    }
```